### PR TITLE
Interactive testing

### PR DESCRIPTION
**Pull request by AlyAtalla**
Used jest framework for testing.
Created a test file(s) ([..].test.js) for a file(s) containing the following functions:
a function for editing the task description.
a function for updating an item's 'completed' status.
the "Clear all completed" function.
Used the mock storage object to mock the localStorage updates.
Mock the HTML elements to test DOM manipulation functions.
 grouped  tests using the describe() method.

### CODE REVIEW

File: src/tests/interactive.test.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    expect(newTask.completed).toBeTruthy();
+  });
+
+  test('Test Clear All Completed Tasks', () => {
+    const task2 = {
+      index: 2,
+      description: 'This is Task 2',
+      completed: true,
+    };
+    const task3 = {
```
Comment (DeliceLydia) [inline src/tests/interactive.test.js:46]: - [x] Please fix tests that are failing
Screenshot 2023-07-19 at 12 18 57

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi Team 

Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Followed git workflow
- No linter errors


## Required Changes ♻️
_Check the comments under the review._


### Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/tests/interactive.test.js (hunk 1 above)
Comment (AlyAtalla) [reply src/tests/interactive.test.js:46]: thank you

**Review by julie-ify — APPROVED** [review]
Hi @AlyAtalla and @Bhekisile,

## Highlights!
- Good code organization and tests ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**